dnf/cli/option_parser.py

```diff
@@ -108,14 +103,25 @@ def __call__(self, parser, namespace, values, opt_str):
             if period != -1:
                 repo = k[:period]
                 k = k[period+1:]
-                if repo not in self.repoopts:
-                    self.repoopts[repo] = argparse.Namespace()
-                setattr(self.repoopts[repo], k, v)
-                setattr(namespace, 'repo_' + self.dest, self.repoopts)
+                if hasattr(namespace, 'repo_setopts'):
+                    repoopts = namespace.repo_setopts
+                else:
+                    repoopts = {}
+                if repo not in repoopts:
+                    repoopts[repo] = {}
+                if k not in repoopts[repo]:
+                    repoopts[repo][k] = []
+                repoopts[repo][k] += [v]
```

Comment:
repoopts[repo].setdefault(k, []).append(v)

Reply:
Thanks for the tip.